pkg/target/target.go

```diff
@@ -108,3 +111,49 @@ func (t *Target) AttackWillLand(a combat.AttackPattern) (bool, string) {
 		return false, "unknown shape"
 	}
 }
+
+func (t *Target) Direction() float64 { return t.direction }
+func (t *Target) SetDirection(trgX, trgY float64) {
+	srcX, srcY := t.Pos()
+	// setting direction to self resets direction
+	if srcX == trgX && srcY == trgY {
+		t.Core.Combat.Log.NewEvent("reset target direction to 0", glog.LogDebugEvent, -1)
+		t.direction = 0
+		return
+	}
+	t.direction = combat.CalcDirection(srcX, srcY, trgX, trgY)
+	t.Core.Combat.Log.NewEvent("set target direction", glog.LogDebugEvent, -1).
+		Write("src target key", t.key).
+		Write("srcX", srcX).
+		Write("srcY", srcY).
+		Write("trgX", trgX).
+		Write("trgY", trgY).
+		Write("direction (in degrees)", combat.DirectionToDegrees(t.direction))
+}
+func (t *Target) SetDirectionToClosestEnemy() {
+	srcX, srcY := t.Pos()
+	// calculate direction towards closest enemy
+	enemyIndex := t.Core.Combat.EnemyByDistance(srcX, srcY, combat.InvalidTargetKey)[0]
```

Comment:
Since I have to refactor everything to use Point anyways, I'll just adjust EnemyByDistance to exclude dead enemies (idk why they are included in the first place, just adds more potential for bugs).

Reply:
Gonna fix this bug later. For now I just refactored it to use `Point`.

Reply:
Will tackle these helper functions in another PR, because this is related to implementing proper targeting for stuff like Amber Q, Ayato Q, Ganyu Q, ...